watertap/costing/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,286 @@
+    blk.cost = pyo.Param(
+        initialize=2.23,
+        units=pyo.units.USD_2021 / pyo.units.kg,  # Anuf et al., 2022
+        mutable=True,
+        doc="Cost of aluminum plate per kg",
+    )
+    costing.register_flow_type("aluminum", blk.cost)
+
+
+def build_iron_cost_param_block(blk):
+    costing = blk.parent_block()
+    blk.cost = pyo.Param(
+        initialize=3.41,
+        units=pyo.units.USD_2021 / pyo.units.kg,  # Anuf et al., 2022
+        mutable=True,
+        doc="Cost of iron plate per kg",
+    )
+    costing.register_flow_type("iron", blk.cost)
+
+
+def build_electrocoagulation_cost_param_block(blk):
+
+    costing = blk.parent_block()
+
+    blk.reactor_capital_cost_base = pyo.Var(
+        initialize=1.15e4,
+        units=pyo.units.USD_2020,
+        doc="Reactor capital cost base cost",
+    )
+
+    blk.reactor_capital_cost_exponent = pyo.Var(
+        initialize=0.45,
+        units=pyo.units.dimensionless,
+        doc="Reactor capital cost exponent",
+    )
+
+    blk.reactor_material_coeff = pyo.Var(
+        initialize=3.4,
+        units=pyo.units.dimensionless,
+        doc="Reactor capital cost material coeff (3.4 for stainless steel; 0.062 for PVC)",
+    )
+
+    blk.reactor_capital_safety_factor = pyo.Var(
+        initialize=2.5,
+        units=pyo.units.dimensionless,
+        doc="Reactor capital cost safety factor",
+    )
+
+    blk.power_supply_capital_slope = pyo.Var(
+        initialize=0.51972,
+        units=pyo.units.USD_2020 / pyo.units.watt,
+        doc="DC power supply + transformer + electrical connection base cost",
+    )
+
+    blk.floc_capital_cost_slope = pyo.Var(
+        initialize=952902,
+        units=pyo.units.USD_2007 / pyo.units.Mgallons,
+        doc="Flocculator capital cost equation slope",
+    )
+
+    blk.floc_capital_cost_intercept = pyo.Var(
+        initialize=177335,
+        units=pyo.units.USD_2007,
+        doc="Flocculator capital cost equation intercept",
+    )
+
+    blk.sludge_handling_cost = pyo.Var(
+        initialize=0,  # 0.025 $/kg from NMSU
+        units=costing.base_currency / pyo.units.kg,
+        doc="Cost per kg for handling generated solids",
+    )
+
+    blk.electrode_material_cost = pyo.Var(
+        initialize=2,
+        units=costing.base_currency / pyo.units.kg,
+        doc="Cost of electrode per kg",
+    )
+
+    blk.electrode_material_cost_safety_factor = pyo.Var(
+        initialize=2,
+        units=pyo.units.dimensionless,
+        doc="Electrode material cost safety factor",
+    )
+
+    blk.number_redundant_reactors = pyo.Var(
+        initialize=2,
+        units=pyo.units.dimensionless,
+        doc="Number of redundant EC reactors",
+    )
+
+    blk.fix_all_vars()
+
+
+@register_costing_parameter_block(
+    build_rule=build_aluminum_cost_param_block,
+    parameter_block_name="aluminum",
+)
+@register_costing_parameter_block(
+    build_rule=build_iron_cost_param_block,
+    parameter_block_name="iron",
+)
```

Comment:
Why register both? Should we just register one depending on whether the use specified aluminum or iron?

Reply:
Ah, right. I was just thinking more about how it might confusing to see anything related to iron costs if you were only considering aluminum, but I suppose it can't hurt considered we've been using the "register everything" approach.